lib/code-path-analysis/code-path-analyzer.js

```diff
@@ -0,0 +1,603 @@
+/**
+ * @fileoverview A class of the code path analyzer.
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assert = require("assert"),
+    CodePath = require("./code-path"),
+    CodePathSegment = require("./code-path-segment"),
+    IdGenerator = require("./id-generator"),
+    debug = require("./debug-helpers");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Gets the last element of a given array.
+ *
+ * @param {any[]} xs - An array to get.
+ * @returns {any|null} The last element, or `null` if the array is empty.
+ */
+function lastOf(xs) {
+    if (xs.length === 0) {
+        return null;
+    }
+    return xs[xs.length - 1];
+}
+
+/**
+ * Checks whether or not a given node is a `case` node.
+ *
+ * @param {ASTNode} node - A `SwitchCase` node to check.
+ * @returns {boolean} `true` if the node is a `case` node.
+ */
+function isCaseNode(node) {
+    return Boolean(node.test);
```

Comment:
I want to write here: `node.test != null` :(
This function's intention is `this node is not "default:".`

Also, I don't like:

```js
if (...) {
    return true;
} else {
    return false;
}
```